beacon-chain/sync/rpc.go

```diff
@@ -128,8 +136,12 @@ func (s *Service) registerRPC(baseTopic string, handle rpcHandler) {
 		// a way to check for its reflect.Kind and based on the result, we can decode
```

Comment:
Why can we get 2 different kinds of values from this topic? This isn't the right PR to hash this out, since for the most part this PR is great and is making things /more/ type safe, but this whole method feels like something isn't right in how p2p is leaking implementation details (primarily through `RPCTopicMappings`) such that the `sync` package needs to be doing reflection on topics.

Reply:
To clarify this is the handler for all our topics, not just one. Depending on the topic type, the expected message type to marshal/unmarshal would be different. If the message type is a pointer/value then we handle them separately.